pkg/executor/insert.go

```diff
@@ -288,7 +340,8 @@ func (e *InsertExec) batchUpdateDupRows(ctx context.Context, newRows [][]types.D
 			if handle == nil {
 				continue
 			}
-			err = e.updateDupRow(ctx, i, txn, r, handle, e.OnDuplicate, updateDupKeyCheck, autoColIdx)
+
+			oldRow, err := getOldRow(ctx, e.Ctx(), txn, r.t, handle, e.GenExprs)
```

Comment:
Similar to the logic above, do we need to check the `oldRow == nil` case?

Reply:
if err == nil && oldRow == nil, it means data corruption @time-and-fate